src/EditorFeatures/CSharp/SplitStringLiteral/SplitStringLiteralCommandHandler.cs

```diff
@@ -125,8 +125,8 @@ private bool SplitString(ITextView textView, ITextBuffer subjectBuffer, int posi
             using var transaction = CaretPreservingEditTransaction.TryCreate(
                 CSharpEditorResources.Split_string, textView, _undoHistoryRegistry, _editorOperationsFactoryService);
 
-            var splitter = StringSplitter.TryCreate(document, position, useTabs, tabSize, indentStyle, cancellationToken);
-            if (splitter?.TrySplit(out var newDocument, out var newPosition) != true)
+            var (newDocument, newPosition) = StringSplitter.TrySplitAsync(document, position, useTabs, tabSize, indentStyle, cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken);
```

Comment:
```suggestion
            var (newDocument, newPosition) = StringSplitter.TrySplitAsync(document, position, useTabs, tabSize, indentStyle, cancellationToken).WaitAndGetResult(cancellationToken);
```

The edit transaction call earlier is absolutely intending for this to be on the UI thread, so the regular method should be called.